### PR TITLE
Webhooks creating forum posts

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
[discord-api-docs#5007](https://github.com/discord/discord-api-docs/pull/5007)
Allows webhooks to create new threads in their forum channel with
```py
await hook.send("## Important Stuff", name="Important Announcement")
```
I'm not too sure about this design though. Maybe something like `hook.create_thread` could be done instead?

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
Create forum post would sound better. Otherwise it could lead to confusion

**Comment by Dorukyum** [comment]
I think we should call this parameter `thread_name`, that should remove the confusion and reflect the API.

**Comment by Lulalaby** [comment]
Correct

**Comment by Middledot** [comment]
So it shoud be `thread_name`?

File: discord/webhook/async_.py [hunk 1]
```diff
@@ -1494,9 +1506,9 @@ async def send(
         ValueError
             The length of ``embeds`` was invalid.
         InvalidArgument
-            There was no token associated with this webhook or ``ephemeral``
-            was passed with the improper webhook type or there was no state
-            attached with this webhook when giving it a view.
+            Either there was no token associated with this webhook, ``ephemeral`` was passed
+            with the improper webhook type, there was no state attached with this webhook when
+            giving it a view, or you specified both ``name`` and ``thread``.
```
Comment (Dorukyum) [inline discord/webhook/async_.py:1511]:
```suggestion
            giving it a view, or you specified both ``thread_name`` and ``thread``.
```

File: discord/webhook/sync.py [hunk 2]
```diff
@@ -1007,10 +1019,11 @@ def send(
         ValueError
             The length of ``embeds`` was invalid
         InvalidArgument
-            There was no token associated with this webhook.
+            There was no token associated with this webhook or you specified both
+            a thread to send to and a thread to create (the ``thread`` and ``name`` parameters).
```
Comment (Dorukyum) [inline discord/webhook/sync.py:1023]:
```suggestion
            a thread to send to and a thread to create (the ``thread`` and ``thread_name`` parameters).
```

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**